multimodal/src/autogluon/multimodal/utils/environment.py

```diff
@@ -70,8 +72,22 @@ def compute_num_gpus(config_num_gpus: Union[int, float, List], strategy: str):
     return num_gpus
 
 
+def get_state_dict_dtype(state_dict):
+    """
+    Returns the first found floating dtype in `state_dict` if there is one, otherwise returns the first dtype.
+    """
+    for t in state_dict.values():
+
+        if t.is_floating_point():
+            return t.dtype
```

Comment:
Do we need this `if`? Can we always return `next(state_dict.values()).dtype`?

Reply:
t can be int